WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift

```diff
@@ -127,6 +134,43 @@ private extension CardReaderSettingsConnectedViewController {
         tableView.reloadData()
     }
 
+    func configureUpdateView() {
+        guard viewModel != nil, let viewModelUpdateInProgress = viewModel?.readerUpdateInProgress else {
+            return
+        }
+        guard updateInProgress != viewModelUpdateInProgress else {
+            return
+        }
+        updateInProgress = viewModelUpdateInProgress
+
+        // If we are not updating a reader, dismiss any updateViewController
+        if !updateInProgress {
+            updateViewController?.dismiss(animated: true, completion: { [weak self] in
+                guard let self = self else {
+                    return
+                }
+                guard let readerUpdateCompletedSuccessfully = self.viewModel?.readerUpdateCompletedSuccessfully else {
+                    return
+                }
+                if readerUpdateCompletedSuccessfully {
+                    self.displayReaderUpdateSuccessNotice()
+                }
+
+                self.updateViewController = nil
+            })
+            return
+        }
+
+        // Otherwise, instantiate and present an updateViewController
+        updateViewController = UpdateViewController()
+        guard updateViewController != nil else {
+            return
+        }
+        updateViewController!.loadViewIfNeeded()
```

Comment:
@ctarda Curiously, my outlets fail to connect unless I do this. Did I create and connect the XIB to the VC incorrectly?

Reply:
Yeah, that's not a good sign, and should not be necessary. Do you have an initializer that calls super.init(nibName:bundle:)?